crates/cargo-wdk/src/actions/build/build_task.rs

```diff
@@ -170,4 +189,108 @@ mod tests {
             &command_exec,
         );
     }
+
+    #[test]
+    fn run_invokes_cargo_build_with_expected_args_and_returns_output() {
+        let working_dir = PathBuf::from("C:/abs/driver");
+        let manifest_path = working_dir.join("Cargo.toml");
+        let manifest_path_string = manifest_path.to_string_lossy().to_string();
+        let profile = Profile::Release;
+        let target_arch = CpuArchitecture::Amd64;
+        let expected_target = super::to_target_triple(target_arch);
```

Comment:
We typically avoid calling the same methods (e.g: `to_target_triple`, `get_cargo_verbose_flags`) used within the method under test (`run`) from the test itself, as this defeats the purpose of the test.

`expected_target` and `flag` should be hard coded expectations, just like all the other expected parameters.